signac/core/jsoncollection.py

```diff
@@ -6,59 +6,97 @@
 This implements the JSON-backend for SyncedCollection API by
 implementing sync and load methods.
 """
-
 import os
 import json
 import errno
 import uuid
 
 from .synced_collection import SyncedCollection
+from .buffered_collection import BufferedSyncedCollection
 from .syncedattrdict import SyncedAttrDict
 from .synced_list import SyncedList
+from .buffered_collection import _get_filemetadata
+
+
+def get_namespace(class_name):
+    """Generate namespace for classname"""
+    return uuid.uuid5(uuid.NAMESPACE_URL, 'signac::'+class_name)
 
 
-class JSONCollection(SyncedCollection):
+class JSONCollection(BufferedSyncedCollection):
     """Implement sync and load using a JSON back end."""
 
     backend = __name__  # type: ignore
 
-    def __init__(self, filename=None, write_concern=False, **kwargs):
-        self._filename = os.path.realpath(filename) if filename is not None else None
-        self._write_concern = write_concern
+    def __init__(self, filename=None, data=None, write_concern=False, no_sync=False, **kwargs):
+        kwargs['data'] = data
         super().__init__(**kwargs)
         if (filename is None) == (self._parent is None):
             raise ValueError(
                 "Illegal argument combination, one of the two arguments, "
                 "parent or filename must be None, but not both.")
+        self.backend_kwargs['filename'] = None if filename is None else os.path.realpath(filename)
+        self.backend_kwargs['write_concern'] = write_concern
+        self.backend_kwargs['backend'] = self.backend
+        self._id = str(uuid.uuid5(get_namespace(type(self).__name__),
```

Comment:
Why are you calling `uuid5` twice? This doesn't make sense to me. The first argument to `uuid5` is a namespace.

Reply:
The first call is to generate name_space and second is to generate id.